src/servers/public-api/books/books.controller.ts

```diff
@@ -19,36 +19,35 @@ import { Logger } from '@providers/logger.provider';
 import { validate } from '@utils/validate.util';
 import { BookDto } from '@servers/public-api/books/books.dto';
 import { services } from '@services/index.service';
+import { FilterBookParams, SortBookParams } from '@interfaces/book.interface';
 
 @Controller('/books')
 export class BooksController {
   private readonly logger = Logger('BooksController');
 
-  @Post('/')
-  async create(
-    @Request() request: ExpressRequest,
-    @Response() response: ExpressResponse,
-    @Next() next: ExpressNextFunction
-  ) {
-    try {
-      const body = await validate<BookDto>(BookDto, request.body);
-      const book = await services.books.registerBook(request.body);
-
-      return response.status(201).json(book);
-    } catch (error) {
-      this.logger.fatal(error);
-      next(error);
-    }
-  }
-
   @Get('/')
   async list(
     @Request() request: ExpressRequest,
     @Response() response: ExpressResponse,
     @Next() next: ExpressNextFunction
   ) {
     try {
-      const book = await services.books.list(request.body.sort, request.body.filter, request);
+      const sort = request.query.sort as SortBookParams;
+      const filter = request.query.filter as FilterBookParams;
+
+      function removeEmptyFields(data: any) {
```

Comment:
This function will be declared for each incoming HTTP request for this resource endpoint.
May want to move it to `books.util.ts` or you can have a middleware that sanitize the query and return an object